supervisor/core/instance_test.go

```diff
@@ -0,0 +1,103 @@
+package core
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const testInstPath = "../../test_instances"
+const testInstName = "test_instance.lua"
+
+// commonStartTest run an Instance and checks its status.
+func commonStartTest(t *testing.T, isSignalIgnore bool) *Instance {
+	assert := assert.New(t)
+
+	instPath := path.Join(testInstPath, testInstName)
+	_, err := exec.LookPath(instPath)
+	assert.Nilf(err, `Unknown instance:"%v".Error: "%v"`, instPath, err)
+
+	// Create an Instance.
+	cmd := exec.Command(instPath)
+	var env []string
+	if isSignalIgnore {
+		env = append(env, "INSTSIGIGNORE=true")
+	}
+	cmd.Env = append(os.Environ(), env...)
+	inst := NewInstance(testInstName, cmd, env, false)
+
+	// Start the Instance.
+	err = inst.Start()
+	assert.Nilf(err, `Can't start an Instance.Error: "%v"`, err)
+
+	// Check status of the Instance.
+	assert.Equal(inst.Status().State, "running",
+		"The Instance is not running.")
+
+	// We need to wait for the new process to set handlers.
+	// It is necessary to update for more correct synchronization.
+	time.Sleep(100 * time.Millisecond)
+
+	return inst
+}
+
+// cleanup sends a SIGKILL signal to all test
+// Instances that remain alive after the test done.
+func cleanupInstanceTest(insts []*Instance) {
+	for _, inst := range insts {
+		if inst.IsAlive() {
+			inst.Cmd.Process.Kill()
+		}
+	}
+}
+
+// A simple test that starts an Instance, terminates it,
+// and checks the status of the Instance.
+func TestInstance(t *testing.T) {
+	assert := assert.New(t)
+	// Set a cleanup callback.
+	var insts []*Instance
+	t.Cleanup(func() { cleanupInstanceTest(insts) })
+
+	inst := commonStartTest(t, false)
+	insts = append(insts, inst)
+
+	err := inst.Stop(100*time.Millisecond, false)
+	assert.Nilf(err, `Can't stop the Instance. Error: "%v"`, err)
+
+	assert.Equal(inst.Status().State, "terminated",
```

Comment:
BTW, what do you think about using string constants for states? Like `stateTerminated`.

Reply:
I don't object.